Updated model diagrams to align with state of specification as of 02-15-24

sbarnum opened the pull request.

sbarnum committed:
Updated model diagrams to align with state of specification as of 02-15-24

kestewart reviewed (changes requested):
On the Core+Software Diagram,   plase remove the WGs and Exmaples part of it.  It contains stale data, and may cause confusion.

kestewart commented:
Also please add your - "Signed-off-by: Sean Barnum <email>" to the comments (should be on the commit, but that's harder to do).

goneall reviewed (changes requested):
A few requested changes

kestewart commented on model Core+Software.png:
packageVerificationCodeExcluededFile was merged.

goneall replied:
Updated in the latest commit

goneall replied:
Updated in the latest commit

goneall commented on model Core+Software.png:
Never mind - it looks like in the diagram we don't use the xsd types

goneall committed:
Update diagrams per feedback

Remove the workgroup and examples
add packageVerificationExcludedFiles
datalicense cardinality
WithAddition updates

Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

goneall commented on model Core+Software.png:
Updated in the latest commit

sbarnum committed:
Removed errant 'algorithm' property on PackageVerificationCode. Signed-off-by: Sean Barnum <sbarnum@mitre.org>

kestewart reviewed (approved):
Changes look good to me.

Sean reviewed and added the last commit. 

Based on this,  and the other commit from Gary,  I'm going to go ahead and merge these.

kestewart reviewed (changes requested):
@goneall - Sean has made it match the text, with his latest commit.   However I think that this was a case where the diagram was correct, and the text should have been fixed.   https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/PackageVerificationCode.md should be updated to allow different algorithms to be specified (ie. SHA256),  right now it's specified as just SHA1.    Issue: https://github.com/spdx/spdx-3-model/issues/646 has been opened to track this.
